qcodes_contrib_drivers/drivers/ERAInstruments/erasynth.py

```diff
@@ -0,0 +1,900 @@
+
+        self.wifi_subnet_address = Parameter(
+            name="wifi_subnet_address",
+            instrument=self,
+            vals=validators.Strings(),
+            get_cmd=f"RA:{_CMD_TO_JSON_MAPPING['PEN']}",
+            set_cmd="PEN{}",
+        )
+        """Sets Subnet mask for WiFi module."""
+
+        self.wifi_gateway_address = Parameter(
+            name="wifi_gateway_address",
+            instrument=self,
+            vals=validators.Strings(),
+            get_cmd=f"RA:{_CMD_TO_JSON_MAPPING['PEG']}",
+            set_cmd="PEG{}",
+        )
+        """Sets default gateway for WiFi module."""
+
+        self.debug_messages_en = Parameter(
+            name="debug_messages_en",
+            instrument=self,
+            vals=validators.Strings(),
+            initial_cache_value=None,
+            val_mapping={True: "1", False: "0"},
+            set_cmd="PD{}",
+        )
+        """Enables/disables debug printing on the serial port."""
+
+        self.visa_handle.baud_rate = BAUDRATE
+        self.timeout(10)  # generous timeout to avoid disrupting measurements
+        self.connect_message()
+        self._prep_communication()
+
+    # ##################################################################################
+    # Public methods
+    # ##################################################################################
+
+    # Standard LO methods
+
+    def on(self) -> None:
+        """
+        Turns ON the RF output.
+        """
+        self.status(True)
+
+    def off(self) -> None:
+        """
+        Turns OFF the RF output.
+        """
+        self.status(False)
+
+    # Custom communication
+
+    def get_idn(self) -> Dict[str, str]:
+        models: Dict[str, str] = {"0": "ERASynth", "1": "ERASynth+", "2": "ERASynth++"}
+        d_status: Dict[str, str] = self.get_diagnostic_status()
+        return {
+            "vendor": "ERA Instruments",
+            "model": models[d_status["model"]],
+            "serial": d_status["serial_number"],
+            "firmware": d_status["em"],
+        }
+
+    def clear_read_buffer(self) -> None:
+        """
+        Clears the read buffer. The instrument often sends information we do not care
+        about in this driver. This function discards the entire read buffer by reading
+        it.
+        """
+        while self.visa_handle.bytes_in_buffer:
+            self.visa_handle.read_bytes(self.visa_handle.bytes_in_buffer)
+
+    def ask(self, cmd: str) -> str:
+        """Writes a command to the communication channel of the instrument and return
+        the response.
+
+        Commands are prefixed with `">"` as required by the ERASynth.
+
+        NB the read buffer is discarded before and after reading one line.
+        """
+        self.clear_read_buffer()
+        response = super().ask(f">{cmd}")
+        self.clear_read_buffer()
+
+        return response
+
+    def ask_raw(self, cmd: str) -> Union[Dict[str, str], str]:
```

Comment:
I am not sure how we can deal with this because mypy expects the type of the return to conform to the type of the `ask_raw` in the superclass, which is `str`.

Reply:
Good point, it is a bit nasty, I will think a bit if I can implement it in some better way 🤔

Reply:
I just double checked (executing the notebook with a debug print of `type(repsponse)`), the output of the function is always `str`, I fixed the type annotation, hope this solves the mypy issue

Reply:
Then I think, `get_diagnostic_status` should be always `str` as commented below?